tests/utils/host_utils.cpp

```diff
@@ -238,7 +238,7 @@ void constructWilsonTestSpinorParam(quda::ColorSpinorParam *cs_param, const Quda
 }
 
 void constructRandomSpinorSource(void *v, int nSpin, int nColor, QudaPrecision precision, QudaSolutionType sol_type,
-                                 const int *const x, quda::RNG &rng)
+                                 const int *const x, int nDim, quda::RNG &rng)
```

Comment:
So this is something that we should file an issue for---for DWF, it's not that meaningful for a source to have non-zero elements anywhere but on the chiral walls. I guess for tests that's less consequential, but we should be careful about that...

Reply:
With this change (without assuming a 4D field) the random field is non-zero for all `s`.

Reply:
Yes, I understand that---my point is that generally when we need a _physical_ random source for DWF it needs to live on the chiral walls. It's fine as is for now, though, for correctness checks and whatnot.

Reply:
Ah gotcha. We can have a separate function, or just a separate knob to handle that in the future.